x/stakeibc/keeper/icqcallbacks_delegator_shares.go

```diff
@@ -42,18 +44,15 @@ func DelegatorSharesCallback(k Keeper, ctx sdk.Context, args []byte, query icqty
 	queriedDelgation := stakingtypes.Delegation{}
 	err := k.cdc.Unmarshal(args, &queriedDelgation)
 	if err != nil {
-		return errorsmod.Wrapf(types.ErrMarshalFailure, "unable to unmarshal query response into Delegation type, err: %s", err.Error())
+		return errorsmod.Wrapf(err, "unable to unmarshal delegator shares query response into Delegation type")
```

Comment:
DelegatorSharesCallback is getting a bit long, hard to understand. Do you think we could break it up?

Reply:
yeah good call, I'll take a peak and see if there's a clean way to do it